src/main/java/net/glowstone/entity/passive/GlowFirework.java

```diff
@@ -0,0 +1,204 @@
+package net.glowstone.entity.passive;
+
+import com.flowpowered.network.Message;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import lombok.Getter;
+import lombok.Setter;
+import net.glowstone.EventFactory;
+import net.glowstone.entity.GlowEntity;
+import net.glowstone.entity.meta.MetadataIndex;
+import net.glowstone.net.message.play.entity.EntityMetadataMessage;
+import net.glowstone.net.message.play.entity.SpawnObjectMessage;
+import net.glowstone.util.InventoryUtil;
+import org.bukkit.EntityEffect;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.SoundCategory;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+import org.bukkit.event.entity.FireworkExplodeEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.bukkit.util.Vector;
+
+public class GlowFirework extends GlowEntity implements Firework {
+
+    @Getter
+    @Setter
```

Comment:
If you are storing the item inside the metadata, no need to keep an instance here. Use the get/set methods for the metadata to store this info (prevents having duplicate instances)